src/main/java/uk/gov/digital/ho/hocs/casework/api/CaseDataTypeService.java

```diff
@@ -0,0 +1,50 @@
+package uk.gov.digital.ho.hocs.casework.api;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.digital.ho.hocs.casework.api.dto.CaseDataType;
+import uk.gov.digital.ho.hocs.casework.client.infoclient.InfoClient;
+import uk.gov.digital.ho.hocs.casework.domain.exception.ApplicationExceptions;
+
+import java.util.Optional;
+import java.util.UUID;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+import static uk.gov.digital.ho.hocs.casework.application.LogEvent.CASE_TYPE_LOOKUP_FAILED;
+
+@Slf4j
+@Service
+public class CaseDataTypeService {
+
+    private final InfoClient infoClient;
+
+    @Autowired
+    public CaseDataTypeService(InfoClient infoClient) {
+        this.infoClient = infoClient;
+    }
+
+    public Stream<CaseDataType> getAllCaseDataTypes(Predicate<CaseDataType> predicate) {
+        return infoClient.getAllCaseDataTypes().stream().filter(predicate);
+    }
+
+    public CaseDataType getCaseDataType(String caseType) {
+        var type = getNonNullValue(caseType);
+        return getAllCaseDataTypes(it -> it.getDisplayCode().equals(type)).findFirst().orElseThrow(() -> getNotFoundException(type));
+    }
+
+    public CaseDataType getCaseDataType(UUID caseUUID){
+        var shortCode = getNonNullValue(caseUUID).toString().substring(34);
```

Comment:
Hmm, can we not move away from the weird manipulation of UUIDs? I know why they are there, it just feels wrong though...

Reply:
I agree, but it needs fixing upstream of here, for the security code we can stop looking up the caseType completely and fetch the actual case from the database, which is quick enough, but then semantically there is a difference 'if it exists this UUID would be a MIN' to 'this case isn't in the database' so we would need to ensure the behaviour stays the same.